bpf/lib/trace.h

```diff
@@ -54,6 +54,7 @@ enum trace_reason {
 	TRACE_REASON_UNKNOWN,
 	TRACE_REASON_SRV6_ENCAP,
 	TRACE_REASON_SRV6_DECAP,
+	TRACE_REASON_ENCRYPT_OVERLAY,
```

Comment:
It seems we've started to misuse this enum because we misnamed it. It shows as `state encrypt-overlay` in the traces :confused: We should probably discuss what we want to do here. Maybe just fix display in `cilium monitor` and Hubble?

Reply:
Hmm, what was the original intent for 'state' ? 
This is a pretty handy spot to just explain 'why' this tracepoint was hit.

Reply:
> Maybe just fix display in cilium monitor and Hubble?

Yes that make sense to me. I think the idea of `trace_reason` being correlated with state doesn't hold anymore since non-CT trace reasons were introduced.

For Hubble there is nothing to update since we don't expose (yet) the trace reason directly (Hubble only infer some other fields from it).

For `cilium monitor` [`TraceNotify.DumpInfo`](https://github.com/cilium/cilium/blob/4082bc38407a3fd2b59f2bd1e8e0e1ed06be8340/pkg/monitor/datapath_trace.go#L274) should be updated and maybe [`TraceNotifyVerbose`](https://github.com/cilium/cilium/blob/4082bc38407a3fd2b59f2bd1e8e0e1ed06be8340/pkg/monitor/datapath_trace.go#L346-L362) depending on how willing we are to break `cilium monitor --json` compatibility.

Reply:
Breaking the json compat is a really good question. I think we need to bring this up in community chat/meeting.

Reply:
@learnitall nooe.

Reply:
Revisiting after seeing the individual commit with the context from the git commit message:

```
    bpf: add trace notification for overlay encryption

    Add a trace notification when we are redirecting a packet back into the
    stack for XFRM encryption.

    Trace example:
    -> stack flow 0xc218244b , identity unknown->unknown state encrypt-overlay ifindex 0 orig-ip 0.0.0.0: 172.18.0.3:58167 -> 172.18.0.2:8472 udp
```

Seeing `-> stack` for a message like this actually makes some sense to me: Basically the packet is being redirected back `to stack`, so it's easier to follow the flow of the packets. The downside is that this means we could end up sending at least two (maybe three?) `TRACE_TO_*` messages per packet, which seems a bit extraneous and may have an adverse impact on CPU usage with Hubble. If I follow correctly though this will only happen if the user lowers the monitor aggregation level below medium. Is that correct?

Looking at `state encrypt-overlay`, I agree it seems a little bit of an odd UX representation. I think the only reasons we emitted trace notifications in a much earlier version was in order to understand the connection tracking state, hence the "reason" field in the datapath only contained CT-based reasons and the CLI presented these as "state". Not sure what Hubble does here. But I agree we can solve this on the CLI UX.

Reply:
> If I follow correctly though this will only happen if the user lowers the monitor aggregation level below medium. Is that correct?

Yes that is the case.

Reply:
Judging from the comments here, and that Alex has already opened an issue for handling the state a bit better, I'm going to resolve this. If there' still an issue here feel free to unresolve and ping me.